api/controller.py

```diff
@@ -1354,6 +1354,12 @@ def fulfill_part_url(part):
                 unicode(e), status_code=e.status_code
             )
 
+        # A subclass of FulfillmentInfo may want to bypass the whole
+        # response creation process.
+        response = fulfillment.as_response
+        if response:
+            return response
+
```

Comment:
Everything below this point assumes that the circ manager can go from a manifest or ACSM link to the manifest or ACSM itself. For Overdrive audio, that's not true.